src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java

```diff
@@ -75,6 +75,9 @@ public void visitClass(String className, InputStream in) {
             // some bug inside ASM causes an IOB exception. Log it and move on?
             // this happens when the class isn't valid.
             logger.warn("Unable to process: " + className);
+        } catch (IllegalArgumentException e) {
+            // [MSHARED-1248] should log instead of failing when analyzing a corrupted jar file
+            logger.warn("Unable to process: " + className, e);
```

Comment:
Make this log message distinct. E.g. "Byte code of " + className + " is corrupt" an possibly include the name or path of the jar file in which the class appears.

Reply:
@elharo 
I updated the message. I do not see where to get a path since we are starting from an InputStream.